swagger/index.yaml

```diff
@@ -0,0 +1,283 @@
+---
```

Comment:
NICE!

A few lessons I learned from flagr is that `required` is hard to deal with, I think we should validate `required` from the application side, and remove the `required` fields. Because the newly added fields cannot be `required`, and it breaks the consistency of swagger schema.

Reply:
Also, the model generated by swagger-gen is also strange - I think 'required' effects whether the item is a pointed or a normal object.   I'll go through and remove those.